pkg/syncer/fullsync.go

```diff
@@ -116,6 +116,101 @@ func CsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer) erro
 		return err
 	}
 
+	if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload &&
+		commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.TKGsHA) {
+		// Replace Volume Metadata using old cluster ID and replace with the new SupervisorID
+		if len(queryAllResult.Volumes) > 0 {
+			var updateMetadataSpecArray []cnstypes.CnsVolumeMetadataUpdateSpec
+			for _, volume := range queryAllResult.Volumes {
+				var UpdatedContainerClusterArray []cnstypes.CnsContainerCluster
+				for _, containercluster := range volume.Metadata.ContainerClusterArray {
+					if containercluster.ClusterId == metadataSyncer.configInfo.Cfg.Global.ClusterID {
+						containercluster.ClusterId = metadataSyncer.configInfo.Cfg.Global.SupervisorID
+					}
+					UpdatedContainerClusterArray = append(UpdatedContainerClusterArray, containercluster)
+				}
+				updateSpecToDeleteMetadata := cnstypes.CnsVolumeMetadataUpdateSpec{
+					VolumeId: cnstypes.CnsVolumeId{
+						Id: volume.VolumeId.Id,
+					},
+					Metadata: cnstypes.CnsVolumeMetadata{
+						ContainerCluster:      volume.Metadata.ContainerCluster,
+						ContainerClusterArray: volume.Metadata.ContainerClusterArray,
+					},
+				}
+
+				updateSpecToAddMetadata := cnstypes.CnsVolumeMetadataUpdateSpec{
+					VolumeId: cnstypes.CnsVolumeId{
+						Id: volume.VolumeId.Id,
+					},
+					Metadata: cnstypes.CnsVolumeMetadata{
+						ContainerCluster:      volume.Metadata.ContainerCluster,
```

Comment:
Can't we set the value to `metadataSyncer.configInfo.Cfg.Global.SupervisorID` directly?

Reply:
done